xocto/ranges.py

```diff
@@ -954,6 +954,37 @@ def is_disjoint(self, other: Range[datetime.date]) -> bool:
             Range(start=other_start, end=other_end, boundaries=other.boundaries)
         )
 
+    @classmethod
+    def parse(cls, range_str: str) -> FiniteDateRange:
```

Comment:
I'm not sure this belongs here to be honest.

Either parsing should be supported for all range types - or serialization/deserialization belongs elsewhere. I'd probably lean towards the latter..

Reply:
> Either parsing should be supported for all range types

Ideally yes, but that's quite a complex problem: to parse arbitrary range boundaries from strings.

> or serialization/deserialization belongs elsewhere

What's the negative impact of having a standard way of deserializing date ranges specifically? 

I'm thinking about this less as de/serialization and more as a kind of pseudo-literal syntax for ranges: it's a shame Python doesn't support ranges as first class literals.

Reply:
I’m thinking it probably doesn’t belong in this library at all